waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/MetaStoreProxyServer.java

```diff
@@ -196,6 +199,29 @@ private void startWaggleDance(
     LOG.info("Waggle Dance has stopped");
   }
 
+  private TProcessorFactory getTProcessorFactory(boolean useSASL) throws TTransportException {
+    if (useSASL) {
+      return new TProcessorFactorySaslDecorator(this.tProcessorFactory, hiveConf);
+    } else {
+      return this.tProcessorFactory;
+    }
+  }
+
+  private TTransportFactory getTTransportFactory(boolean useFramedTransport, boolean useSASL)
+    throws TTransportException, LoginException {
+    TTransportFactory transFactory;
+    if (useFramedTransport) {
+      transFactory = new TFramedTransport.Factory();
+    } else {
+      transFactory = new TTransportFactory();
+    }
+    if (useSASL) {
+      UserGroupInformation.setConfiguration(hiveConf);
+      transFactory = new HiveAuthFactory(hiveConf).getAuthTransFactory();
```

Comment:
Does this mean that if `useSASL` is `true` that the value for `useFrameTransport` is effectively ignored as the value of `transFactory` is reset here regardless of what happens in lines 213-217.